google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITBucketTest.java

```diff
@@ -417,6 +419,70 @@ public void testCreateBucketWithAutoclass() {
     }
   }
 
+  @Test
+  @CrossRun.Exclude(transports = Transport.GRPC)
+  public void testObjectRetention() {
+    String bucketName = generator.randomBucketName();
+
+    // Create a bucket with object retention enabled
+    storage.create(
+        BucketInfo.newBuilder(bucketName).build(), BucketTargetOption.enableObjectRetention(true));
+
+    try {
+      Bucket remoteBucket = storage.get(bucketName);
```

Comment:
If interested, you could use the TemporaryBucket helper to create and automatically cleanup the bucket. Example of this https://github.com/googleapis/java-storage/blob/cf944aeb9f8be34c7e860a57515971b626267cb6/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITBucketTest.java#L123-L124